src/ceph-disk/ceph_disk/main.py

```diff
@@ -3756,6 +3756,25 @@ def main_activate(args):
             )
             osd_data = get_mount_point(cluster, osd_id)
 
+            args.cluster = cluster
+
+            # Check if encrypted block device in journal
+
+            block_dmcrypt_path = os.path.join(osd_data, 'block_dmcrypt')
+
+            if args.dmcrypt and os.path.exists(block_dmcrypt_path):
+                partition = DevicePartition.factory(
+                    path=None,
+                    dev=os.path.join(osd_data, 'block_dmcrypt'),
+                    args=args)
+
+                partition.osd_dm_key = get_dmcrypt_key(
```

Comment:
> I tried fixing mount_activate instead. However you need access to the args dictionary or at least the relevant entries for some of the functions being called. We would need to manually build that args array then.

ah, i see now. `DevicePartition.factory()` needs the `args` dict. so probably we can have the same fix in `main_activate_space()`?